### PR TITLE
Pass request_id onto the task if called from a task

### PR DESCRIPTION
**Pull request by servingUpAces**
If the request_id exists in the Flask global context, add it to the kwargs for the task.

The request_id is set is the task is created from a http request, if that task then calls through to another task this will set the request_id from the global context. We should then be able to follow the creation of a notification all the way from the original http request to the sending task.

### CODE REVIEW

**Review by idavidmcdonald — APPROVED** [review]
Would be nice if there was an easy test to support it but it's not the end of the world

**Comment by servingUpAces** [comment]
I agree

File: app/celery/celery.py [hunk 1]
```diff
@@ -58,6 +58,8 @@ def apply_async(self, args=None, kwargs=None, task_id=None, producer=None,
 
             if has_request_context() and hasattr(request, 'request_id'):
                 kwargs['request_id'] = request.request_id
+            elif g.request_id:
```
Comment (idavidmcdonald) [inline app/celery/celery.py:61]: Oh, will this raise an attribute error if there is g does not have `request_id` as an attribute?

I just opened a `flask shell` and did `g.thing` and got an attribute error so wondering if we are at risk of the same here?

File: app/celery/celery.py [hunk 2]
```diff
@@ -58,6 +57,11 @@ def apply_async(self, args=None, kwargs=None, task_id=None, producer=None,
 
             if has_request_context() and hasattr(request, 'request_id'):
                 kwargs['request_id'] = request.request_id
+            else:
```
Comment (idavidmcdonald) [inline app/celery/celery.py:60]: Optional: What you've written is all good I think it looks a bit cleaner as the following

```
elif g.hasattr("request_id") and g.request_id:
    kwargs['request_id'] = g.request_id
```

But as I said, all fine so you can pick what you like most

**Review by idavidmcdonald — APPROVED** [review]

File: app/celery/celery.py (hunk 2 above)
Comment (servingUpAces) [reply app/celery/celery.py]: g doesn't have hasattr(), but I've taken the same approach as notifications-utils

**Pull request merged**